src/room/AppSelectionModal.tsx

```diff
@@ -50,12 +50,11 @@ export const AppSelectionModal: FC<Props> = ({ roomId }) => {
         ? window.location.href
         : getRoomUrl(roomId, roomSharedKey ?? undefined)
     );
-    // Edit the URL so that it opens in embedded mode. We do this for two
-    // reasons: It causes the mobile app to limit the user to only visiting the
-    // room in question, and it prevents this app selection prompt from being
-    // shown a second time.
+    // Edit the URL to prevent the app selection prompt from appearing a second
+    // time within the app, and to keep the user confined to the current room
     url.hash = editFragmentQuery(url.hash, (params) => {
-      params.set("embed", "");
+      params.set("appPrompt", "false");
+      params.set("confineToRoom", "");
```

Comment:
What is the advantage of making `"addPrompt"` a boolean while `"confinedToRoom"`, `"hideHeader"` ... just toggle by either beeing part of the url or now (this is what I mean: `...=prevVla&hideHeader=&nextprop...`)

Reply:
Good point, they can and should be handled the same way